tests/PHPStan/Rules/Pure/PureMethodRuleTest.php

```diff
@@ -167,4 +177,16 @@ public function testImpureAssignRef(): void
 		]);
 	}
 
+	public function testBug11207a(): void
+	{
+		$this->treatPhpDocTypesAsCertain = false;
+		$this->analyse([__DIR__ . '/data/bug-11207.php'], []);
+	}
+
+	public function testBug11207b(): void
+	{
+		$this->treatPhpDocTypesAsCertain = true;
+		$this->analyse([__DIR__ . '/data/bug-11207.php'], []);
+	}
```

Comment:
I think given example should not produce purity errors no matter `treatPhpDocTypesAsCertain`, as we are dealing with a non-explicit never.

therefore I added both tests, even though they actually have the same assertions. does this make sense?